src/thunderbots/software/gui/widgets/main_widget.cpp

```diff
@@ -0,0 +1,464 @@
+    setupGameStateOverrideComboBox();
+    setupPlayOverrideComboBox();
+}
+
+void MainWidget::setupAIStartAndStopButtons()
+{
+    auto start_ai_func = []() { Util::DynamicParameters::AI::run_ai.setValue(true); };
+    connect(main_widget->start_ai_button, &QPushButton::clicked, start_ai_func);
+    auto stop_ai_func = []() { Util::DynamicParameters::AI::run_ai.setValue(false); };
+    connect(main_widget->stop_ai_button, &QPushButton::clicked, stop_ai_func);
+}
+
+void MainWidget::setupTeamColourComboBox()
+{
+    main_widget->team_colour_combo_box->insertItem(0, "Yellow");
+    main_widget->team_colour_combo_box->insertItem(1, "Blue");
+    main_widget->team_colour_combo_box->insertItem(2, "Use Refbox");
+    auto on_team_colour_changed = [](const QString& text) {
+        if (text == "Yellow")
+        {
+            Util::DynamicParameters::AI::refbox::override_refbox_friendly_team_color
+                .setValue(true);
+            Util::DynamicParameters::AI::refbox::friendly_color_yellow.setValue(true);
+        }
+        else if (text == "Blue")
+        {
+            Util::DynamicParameters::AI::refbox::override_refbox_friendly_team_color
+                .setValue(true);
+            Util::DynamicParameters::AI::refbox::friendly_color_yellow.setValue(false);
+        }
+        else if (text == "Use Refbox")
+        {
+            Util::DynamicParameters::AI::refbox::override_refbox_friendly_team_color
+                .setValue(false);
+        }
+        else
+        {
+            LOG(FATAL) << "Tried to set the team colour with an invalid value: '"
+                       << text.toStdString() << "'" << std::endl;
+        }
+    };
+    connect(main_widget->team_colour_combo_box, &QComboBox::currentTextChanged,
+            on_team_colour_changed);
+}
+
+void MainWidget::setupGameStateOverrideComboBox()
+{
+    // TODO: Set this up using factory values like the play override once a factory of
+    // these values is available
+    main_widget->gamestate_override_combo_box->insertItem(0, "None");
+    main_widget->gamestate_override_combo_box->insertItem(1, "Play");
+    main_widget->gamestate_override_combo_box->insertItem(2, "Halt");
+    main_widget->gamestate_override_combo_box->insertItem(3, "Stop");
+
+    auto on_gamestate_changed = [](const QString& text) {
+        if (text == "Use Refbox")
+        {
+            Util::DynamicParameters::AI::refbox::override_refbox_play.setValue(false);
+        }
+        else
+        {
+            Util::DynamicParameters::AI::refbox::override_refbox_play.setValue(true);
+            Util::DynamicParameters::AI::refbox::current_refbox_play.setValue(
+                text.toStdString());
+        }
+    };
+    connect(main_widget->gamestate_override_combo_box, &QComboBox::currentTextChanged,
+            on_gamestate_changed);
+}
+
+void MainWidget::setupDefendingSideComboBox()
+{
+    main_widget->defending_side_combo_box->insertItem(0, "Use Refbox");
```

Comment:
Can we make a note here linking it to this issue? https://github.com/UBC-Thunderbots/Software/issues/811

Reply:
done